lib/redis/store.rb

```diff
@@ -42,8 +42,14 @@ def reconnect
     end
 
     def to_s
-      h = @client.host
-      "Redis Client connected to #{/:/ =~ h ? '['+h+']' : h}:#{@client.port} against DB #{@client.db}"
+      if @client.path
+        h = @client.path
+      else
+        h = @client.host
+        h = "[#{h}]" if h.include?(":")
+        h = "#{h}:#{@client.port}"
+      end
+      "Redis Client connected to #{h} against DB #{@client.db}"
```

Comment:
This method really should be delegating the URL generation to another method, instead of doing it all in the `to_s` here.

Reply:
I could use Redis::Client#id. it's not as good as this one (it returns a non-usable URI for Unix sockets) but it pushes the responsibility upstream. you want me to do that?

Reply:
no, I want you to create another method handling the logic for generating a Redis Server URL, like `Redis::Store#url` or something.

Reply:
okey dokey, new method (`Redis::Store#location`) added instead. new new PS pushed